src/sotoki/utils/generator.py

```diff
@@ -23,9 +23,15 @@ def run(self):
         # parse XML file. not using defusedxml for performances reasons.
         # although containing user-generated content, we trust Stack Exchange dump
         parser = xml.sax.make_parser()  # nosec
-        parser.setContentHandler(self.walker(processor=self.processor_callback))
-        parser.parse(self.fpath)
-        parser.setContentHandler(None)
+        try:
+            parser.setContentHandler(self.walker(processor=self.processor_callback))
+            parser.parse(self.fpath)
+            parser.setContentHandler(None)
+        finally:
+            try:
+                parser.close()
+            except xml.sax.SAXException as e:
```

Comment:
Would you mind use `exc` as variable name and logger.exception to respect convention and get the trace?

```py
except xml.sax.SAXException as exc:
    logger.exception(exc)
```